js/apps/admin-ui/src/realm-settings/OpenIdEndpointConigurationTab.tsx

```diff
@@ -0,0 +1,175 @@
+  save: (realm: RealmRepresentation) => void;
+};
+
+export const OpenIdEndpointConigurationTab = ({
+  realm,
+  save,
+}: RealmSettingsThemesTabProps) => {
+  const { t } = useTranslation();
+
+  const [newClaim, setNewClaim] = useState<string>("");
+  const { control, handleSubmit, setValue } = useForm<RealmRepresentation>();
+
+  const setupForm = () => {
+    let attributes = realm.attributes || {};
+    
+    if (!("claimsSupported" in attributes)) {
+      getOpenIdEndpointConfiguration().then((result) => {
+        attributes.claimsSupported = result.claims_supported.join(",");
+        setValue("attributes", attributes);
+      });
+    }
+    else{
+      setValue("attributes", attributes);
+    }
+    
+  };
+
+  useEffect(setupForm, []);
+
+  async function getOpenIdEndpointConfiguration(): Promise<any> {
+    const response = await fetch(
+      `${addTrailingSlash(adminClient.baseUrl)}/realms/${
+        realm.realm
+      }/.well-known/openid-configuration`,
+      {
+        method: "GET",
+        headers: getAuthorizationHeaders(await adminClient.getAccessToken()),
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error(
+        `Server responded with invalid status: ${response.statusText}`,
+      );
+    }
+    return response.json();
+  }
+
+  return (
+    <PageSection variant="light">
+      <FormAccess
+        isHorizontal
+        role="manage-realm"
+        className="pf-u-mt-lg"
+        onSubmit={handleSubmit(save)}
+      >
+        <FormGroup
+          label={t("claimsSupported")}
+          fieldId="kc-claims-supported"
+          labelIcon={
+            <HelpItem
+              helpText={t("claimsSupportedHelp")}
+              fieldLabelId="claimsSupported"
+            />
+          }
+        >
+          <Controller
+            name="attributes"
+            control={control}
+            render={({ field }) => {
+              const claimsSupported = field.value?.claimsSupported?.trim()
+                ? field.value.claimsSupported.split(",")
+                : [];
+              return (
+                <>
+                  {claimsSupported.map((claim: string, index: number) => {
+                    if (claim) {
+                      return (
+                        <InputGroup key={index}>
+                          <TextInput
+                            id={"textInput" + index}
+                            value={claim}
+                            onChange={(value) => {
+                              claimsSupported[index] = value;
+                              field.value &&
+                                (field.value.claimsSupported =
+                                  claimsSupported.join(","));
+                              field.onChange(field.value);
+                            }}
+                            aria-label="input with button"
+                          />
+                          <Button
+                            id={"inputButtonRemove" + index}
```

Comment:
If this `id` is not used it can be removed. If it's needed for the tests we prefer using a test id.
```suggestion
                            data-testid={"inputButtonRemove" + index}
```